sycl/include/sycl/ext/intel/experimental/esimd/memory.hpp

```diff
@@ -783,7 +786,10 @@ __ESIMD_API
                __ESIMD_NS::simd_mask<N> pred = 1) {
 #ifdef __ESIMD_FORCE_STATELESS_MEM
   return lsc_gather<T, NElts, DS, L1H, L2H>(
-      reinterpret_cast<T *>(acc.get_pointer().get()), offsets, pred);
+      reinterpret_cast<std::add_pointer_t<sycl::detail::copy_cv_qualifiers_t<
```

Comment:
Note for reviewers: `accessor::get_pointer()` is deprecated for device side accessors in SYCL 2020 and `get_multi_ptr()` should be used instead. `get_multi_ptr()` uses `acessor::value_type` as pointee type and that in turn can be qualified as `const` if accessor is read-only.

Deprecated `accessor::get_pointer()` was updated to also use `accessor::value_type` in #13443 to match the upcoming spec change (previous behavior is treated as an oversight in the spec) and therefore it triggered an issue where `reinterpret_cast` would cast away `const` qualifier.

This is why I've changed the API we are using here and why that machinery of copying CV qualifiers is needed.